servant-client/test/Servant/SuccessSpec.hs

```diff
@@ -129,6 +130,14 @@ successSpec = beforeAll (startWaiApp server) $ afterAll endWaiApp $ do
         Left e -> assertFailure $ show e
         Right val -> getHeaders val `shouldBe` [("X-Example1", "1729"), ("X-Example2", "eg2")]
 
+    it "Returns headers on UVerb requests" $ \(_, baseUrl) -> do
+      res <- runClient getUVerbRespHeaders baseUrl
+      case res of
+        Left e -> assertFailure $ show e
+        Right (Z (I (WithStatus val))) ->
+          getHeaders val `shouldBe` [("X-Example1", "1729"), ("X-Example2", "eg2")]
+        Right (S _) -> assertFailure "expected first alternative of union"
```

Comment:
There are helpers for this, too: https://hackage.haskell.org/package/servant-0.18.2/docs/Servant-API-UVerb-Union.html#v:foldMapUnion, https://hackage.haskell.org/package/servant-0.18.2/docs/Servant-API-UVerb-Union.html#v:matchUnion

Reply:
There is a similar inference problem here. Is this better?
```haskell
      case res of
        Left e -> assertFailure $ show e
        Right val -> case matchUnion val of
          Just (WithStatus val' :: WithStatus 200 (Headers TestHeaders Bool))
            -> getHeaders val' `shouldBe` [("X-Example1", "1729"), ("X-Example2", "eg2")]
          Nothing -> assertFailure "expected first alternative of union"
```